bindings/nodejs/src/layers.rs

```diff
@@ -0,0 +1,142 @@
+use napi::bindgen_prelude::*;
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::time::Duration;
+use std::usize;
+
+/// Layer is the mechanism to intercept operations.
+#[derive(PartialEq, Eq)]
+#[napi]
+pub enum Layer {
+    /// Add concurrent request limit.
+    ///
+    /// Users can control how many concurrent connections could be established between OpenDAL and underlying storage services.
+    ///
+    /// - @param {number} permits
+    ///
+    /// Create a new ConcurrentLimitLayer will specify permits
+    ConcurrentLimit,
+
+    /// Add an immutable in-memory index for underlying storage services.
+    ///
+    /// Especially useful for services without list capability like HTTP.
+    ///
+    /// - @param {Array<string>} keys
+    ///
+    /// The keys which will be inserted into the index.
+    ///
+    /// Please use `JSON.stringfy` to serialize because of type restrictions.
+    ImmutableIndex,
```

Comment:
I don't think this layer is worth export. Let's remove it.

Reply:
Currently, it is not possible to directly export a struct containing a layer to be used by the JavaScript side. 
I am exporting this enum in order to differentiate between layers. 
Using other types, such as string, to specify the layer type doesn't seem very elegant. After all, the name of the layer type is quite long.🤔